### PR TITLE
Ignore clippy warnings and fix functions with arrays

### PR DESCRIPTION
**Pull request by EPashkin**
Apply https://github.com/gtk-rs/gir/pull/609 and https://github.com/gtk-rs/gir-files/pull/23

cc @GuillaumeGomez , @sdroege

### CODE REVIEW

File: gobject-sys/src/lib.rs [hunk 1]
```diff
@@ -1589,7 +1590,7 @@ extern "C" {
     pub fn g_object_get_qdata(object: *mut GObject, quark: glib::GQuark) -> gpointer;
     //pub fn g_object_get_valist(object: *mut GObject, first_property_name: *const c_char, var_args: /*Unimplemented*/va_list);
     #[cfg(any(feature = "v2_54", feature = "dox"))]
-    pub fn g_object_getv(object: *mut GObject, n_properties: c_uint, names: *mut c_char, values: GValue);
+    pub fn g_object_getv(object: *mut GObject, n_properties: c_uint, names: *mut *const c_char, values: *mut GValue);
```
Comment (sdroege) [inline gobject-sys/src/lib.rs]: This changes quite a bit

New gir-files or what is causing this?

**Comment by sdroege** [comment]
Looks good to me

File: gobject-sys/src/lib.rs (hunk 1 above)
Comment (GuillaumeGomez) [reply gobject-sys/src/lib.rs]: I have concerns about this change as well.

Comment (EPashkin) [reply gobject-sys/src/lib.rs]: Yes, it was added before in https://github.com/gtk-rs/sys/pull/93 as manual functions.

**Comment by EPashkin** [comment]
CI passed (with one bad on travis due futures)

**Comment by GuillaumeGomez** [comment]
Then let's merge. Thanks!

**Pull request merged**